### PR TITLE
BCI-4128: Use Starknet-Foundry

### PR DESCRIPTION
**Pull request by augustbleeds**
Uses Starknet foundry to run the tests in the main chainlink package.

- [x] update examples package to 0.27.0
- [x] introduce starknet-foundry to main package and refactor tests
- [x] update examples starknet-foundry action to 0.27.0
- [x] replace default scarb test command to use `snforge test`
- [x] installs starknet-foundry for contract gh workflow tests

### CODE REVIEW

File: contracts/Scarb.lock [hunk 1]
```diff
@@ -1,14 +1,74 @@
 # Code generated by scarb DO NOT EDIT.
 version = 1
 
+[[package]]
```
Comment (augustbleeds) [inline contracts/Scarb.lock:4]: adding this to make sure alexandria package compiles for mcms

File: examples/contracts/aggregator_consumer/Scarb.toml [hunk 2]
```diff
@@ -8,10 +8,13 @@ name = "aggregator_consumer"
 version = "0.1.0"
 cairo-version = "2.6.3"
 
+# [scripts]
+# test = "snforge test"
+
```
Comment (chris-de-leon-cll) [inline examples/contracts/aggregator_consumer/Scarb.toml:13]: We probably don't need this either right?

File: contracts/src/tests/test_aggregator.cairo [hunk 3]
```diff
@@ -139,10 +142,13 @@ fn test_access_control() {
...
-        Aggregator::TEST_CLASS_HASH.try_into().unwrap(), 0, calldata.span(), false
-    )
-        .unwrap();
+
+    let (aggregatorAddr, _) = declare("Aggregator").unwrap().deploy(@calldata).unwrap();
+
+    // let (aggregatorAddr, _) = deploy_syscall(
+    //     Aggregator::TEST_CLASS_HASH.try_into().unwrap(), 0, calldata.span(), false
+    // )
+    //     .unwrap();
```
Comment (chris-de-leon-cll) [inline contracts/src/tests/test_aggregator.cairo:151]: We can remove this right?

**Review by chris-de-leon-cll — APPROVED** [review]

File: examples/contracts/aggregator_consumer/Scarb.toml (hunk 2 above)
Comment (augustbleeds) [reply examples/contracts/aggregator_consumer/Scarb.toml:13]: yeah, we don't. i might add it though to make use of `scarb test`

**Pull request merged**